includes/class-wc-admin-notes-order-milestones.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * WooCommerce Admin (Dashboard) Order Milestones Note Provider.
+ *
+ * Adds a note to the merchant's inbox when certain order milestones are reached.
+ *
+ * @package WooCommerce Admin
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * WC_Admin_Notes_Order_Milestones
+ */
+class WC_Admin_Notes_Order_Milestones {
+	/**
+	 * Name of the "first order" note.
+	 */
+	const FIRST_ORDER_NOTE_NAME = 'wc-admin-first-order';
+
+	/**
+	 * Name of the "ten orders" note.
+	 */
+	const TEN_ORDERS_NOTE_NAME = 'wc-admin-ten-orders';
+
+	/**
+	 * Name of the "other milestones" note.
+	 */
+	const ORDERS_MILESTONE_NOTE_NAME = 'wc-admin-orders-milestone';
+
+	/**
+	 * Option key name to store last order milestone.
+	 */
+	const LAST_ORDER_MILESTONE_OPTION_KEY = 'wc_admin_last_orders_milestone';
+
+	/**
+	 * Hook to process order milestones.
+	 */
+	const PROCESS_ORDERS_MILESTONE_HOOK = 'wc_admin_process_orders_milestone';
+
+	/**
+	 * Allowed order statuses for calculating milestones.
+	 *
+	 * @var array
+	 */
+	protected $allowed_statuses = array(
```

Comment:
I feel like we might have chatted about this in slack, but did we want to perhaps make this list filter-able? Perhaps I get too filter happy, but it seems like there are numerous extensions that allow manipulation of order statuses.

Reply:
That's a good point about extensions - I'll make this filterable.

Reply:
Added a filter in 90a1c346.